src/impl.ml

```diff
@@ -992,7 +983,11 @@ let serve common_options source source_fd source_format source_protocol destinat
         | _ -> failwith (Printf.sprintf "Not implemented: serving from source %s" source) ) >>= fun source_sock ->
       ( match destination_endpoint with
         | File path ->
-          open_with_creat path true >>= fun fd ->
+          ( if not(Sys.file_exists path) then begin
+              Lwt_unix.openfile path [ Unix.O_CREAT; Unix.O_RDONLY ] 0o0644 >>= fun fd ->
+              Lwt_unix.close fd
+            end else return () ) >>= fun () ->
+          Vhd_format_lwt.IO.openfile path true >>= fun fd ->
```

Comment:
I might've reverted too much of your patch here.
Would be better to `Lwt_unix.openfile path [ Unix.O_RDWR; Unix.O_CREAT ] 0o0644`? directly?

Reply:
Unfortunately that will change the types everywhere - we could do this as an optimisation in a separate PR.